payroll-processor-client/src/app/employee/employee-list/employee-list.component.ts

```diff
@@ -1,30 +1,38 @@
-import { Component, OnInit, OnDestroy } from '@angular/core';
-import { Employee } from '../../data/state/employee-model';
-import { DataService } from 'src/app/data/data-service';
-import { faSmileBeam, faSkull } from '@fortawesome/free-solid-svg-icons';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { faSkull, faSmileBeam } from '@fortawesome/free-solid-svg-icons';
+
+import { Employee } from './state/employee-list.model';
+import { EmployeeListQuery } from './state/employee-list.query';
+import { EmployeeListService } from './state/employee-list.service';
 
 @Component({
   selector: 'app-employee-list',
   templateUrl: './employee-list.component.html',
   styleUrls: ['./employee-list.component.scss'],
 })
 export class EmployeeListComponent implements OnInit, OnDestroy {
-  faSkull = faSkull;
-  faSmileBeam = faSmileBeam;
+  readonly faSkull = faSkull;
+  readonly faSmileBeam = faSmileBeam;
+  readonly employees: Observable<Employee[]>;
 
-  constructor(private dataService: DataService) {}
+  constructor(
+    private service: EmployeeListService,
+    private query: EmployeeListQuery,
+  ) {
+    this.employees = this.query.selectAll();
+  }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.service.getEmployees().subscribe();
```

Comment:
You can do this in the constructor instead of `ngOnInit()`.

The best way of thinking about the 2 is `constructor` is for everything to prep the component independent of DOM interactions.

`ngOnInit()` is a lifecycle hook specifically for 'Angular' things, like DOM interactions and manipulation.

Since `this.service.getEmployees().subscribe()` doesn't touch the DOM or other 'Angular' things, you are fine to put it in the constructor.

Reply:
![image](https://user-images.githubusercontent.com/11415127/80320478-9cbac300-87e4-11ea-93be-11ba3b968e76.png)

Reply:
This is what I feel like you are really telling me with these comments 😆 

![image](https://user-images.githubusercontent.com/11415127/80320493-b3611a00-87e4-11ea-8674-a1f9d500c48f.png)

Reply:
haha no way! The team I'm working with now has been doing Angular for 4-5 years and I leave these comments on their PRs.

There's a lot of nuance to all this stuff. It's not easy.